akka-actor-typed/src/main/scala/akka/actor/typed/internal/InternalMessageChannel.scala

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal
+
+import akka.actor.ActorRefProvider
+import akka.actor.typed.MessageChannel
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] trait InternalMessageChannel[-T] extends MessageChannel[T] {
+
+  /**
+   * Get a reference to the actor ref provider which created this ref.
+   */
+  def provider: ActorRefProvider
+
+  /**
+   * @return `true` if the actor is locally known to be terminated, `false` if alive or uncertain.
+   */
+  def isTerminated: Boolean
+
+}
```

Comment:
Could we avoid this internal extra-trait and have one that is `Aliveness` (or better name) that we only implement where it can be true and match on that in ask instead of implementing this with a constant `false` return in a bunch of places? Or maybe just put it in `MessageChannel` with a default `false` instead?

Reply:
ok, I'll add such a separate internal trait, which `EntityRef` and other non `ActorRef` doesn't have to implement. Not implementing it means `false` (don't know)